dandi/publish/views/draft_version.py

```diff
@@ -0,0 +1,80 @@
+from django.contrib.auth.models import User
+from django.shortcuts import get_object_or_404
+from rest_framework import serializers, status
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+from rest_framework_extensions.mixins import DetailSerializerMixin, NestedViewSetMixin
+
+from dandi.publish.models import Dandiset, DraftVersion
+from dandi.publish.tasks import publish_version
+from dandi.publish.views.dandiset import DandisetSerializer
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = [
+            'email',
+            'username',
+        ]
+
+
+class DraftVersionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = DraftVersion
+        fields = [
+            'dandiset',
+            'name',
+            'created',
+            'modified',
+            'locked',
+            'locked_by',
+        ]
+        read_only_fields = ['created']
+
+    dandiset = DandisetSerializer()
+    locked_by = UserSerializer()
+
+
+class DraftVersionDetailSerializer(DraftVersionSerializer):
+    class Meta(DraftVersionSerializer.Meta):
+        fields = DraftVersionSerializer.Meta.fields + ['metadata']
+
+
+class DraftVersionViewSet(NestedViewSetMixin, DetailSerializerMixin, GenericViewSet):
```

Comment:
This is probably more semantically appropriate as a https://www.django-rest-framework.org/api-guide/views/